### PR TITLE
Implement GithubStorage for LedgerManager (read)

### PR DESCRIPTION
**Pull request by hammadj**
# Description

A GithubStorage class for the LedgerManager would allow reading and writing
the ledger.json file directly to Github programmatically without having to
go through manual git commits. Useful to automate instance maintainer tasks
and in the future enable self-service admin.

Related: #2450

# Test Plan

- Build the NPM package and import it in an external JS file
- create a GithubStorage instance and call the `read()` method
- ensure the ledger is successfully fetched from the given cred instance

### CODE REVIEW

File: src/api/ledgerStorage/github.js
Comment (topocount) [inline src/api/ledgerStorage/github.js:24]: From my testing last month, the V4 GraphQL API is probably not going to work with larger ledgers that are already in production. I did some testing with the 1Hive ledger and found that the returned result was consistently truncated.

The V3 REST API has much higher payload limits if we use the lower-level git endpoints.

I'll do some more stress testing via cURL and update using this query specifically.

**Review by topocount — COMMENTED** [review]
@hammadj do you recommend testing this in node? I'm having issues utilizing this and I believe it's stemming from the `cross-fetch` import:
```node
> let test = new storage.GithubStorage(token, "sourcecred/cred")
> test.read()
Promise { <pending> }
> (node:57295) UnhandledPromiseRejectionWarning: ReferenceError: XMLHttpRequest is not defined
    at /home/kevin/Projects/sourceCred/sc-core/dist/api.js:8:91878
    at new Promise (<anonymous>)
    at E (/home/kevin/Projects/sourceCred/sc-core/dist/api.js:8:91745)
    at _s.read (/home/kevin/Projects/sourceCred/sc-core/dist/api.js:18:110128)
    at REPL400:1:6
```

File: src/api/ledgerStorage/github.js
Comment (topocount) [reply src/api/ledgerStorage/github.js:2]: I tried explicitly importing cross-fetch here and got an error on importing `api.js` so it seems like we need to find a better way to configure this in webpack? I'm running node v14.15.4 by the way.

**Review by topocount — APPROVED** [review]
I'm able to get this working in node after rebasing on #2711.

Granting approval to merge this after that gets merged.

File: src/api/ledgerStorage/github.js
Comment (hammadj) [reply src/api/ledgerStorage/github.js:2]: No, it might be used in the Browser for local instance administration

File: src/api/ledgerStorage/github.js
Comment (hammadj) [reply src/api/ledgerStorage/github.js:23]: This was actually a workaround since the GraphQL API didn't support fetching blobs larger than 1MB 😅

**Pull request merged**
